docs/source/main-algorithms/froidure-pin/class.rst

```diff
@@ -0,0 +1,77 @@
+.. Copyright (c) 2024 J. D. Mitchell
+
+   Distributed under the terms of the GPL license version 3.
+
+   The full license is in the file LICENSE, distributed with this software.
+
+.. currentmodule:: libsemigroups_pybind11
+
+The FroidurePin class
+=====================
+
+.. autoclass:: FroidurePin
+   :doc-only:
+   :class-doc-from: class
+
+Contents
+--------
+
+.. autosummary::
+   :signatures: short
+   
+   FroidurePin.add_generator
+   FroidurePin.add_generators
+   FroidurePin.batch_size
+   FroidurePin.closure
+   FroidurePin.contains
+   FroidurePin.contains_one
+   FroidurePin.copy
+   FroidurePin.copy_add_generators
+   FroidurePin.copy_closure
+   FroidurePin.current_left_cayley_graph
+   FroidurePin.current_length 
+   FroidurePin.current_max_word_length
+   FroidurePin.current_number_of_rules
+   FroidurePin.current_position
+   FroidurePin.current_right_cayley_graph
+   FroidurePin.current_size
+   FroidurePin.currently_contains_one
+   FroidurePin.degree
+   FroidurePin.enumerate
+   FroidurePin.fast_product
+   FroidurePin.final_letter
+   FroidurePin.first_letter
+   FroidurePin.generator
+   FroidurePin.idempotents
+   FroidurePin.init
+   FroidurePin.is_finite
+   FroidurePin.is_idempotent
+   FroidurePin.left_cayley_graph
+   FroidurePin.length
+   FroidurePin.number_of_elements_of_length
+   FroidurePin.number_of_generators
+   FroidurePin.number_of_idempotents
+   FroidurePin.number_of_rules
+   FroidurePin.position
+   FroidurePin.prefix
+   FroidurePin.reserve
+   FroidurePin.right_cayley_graph
+   FroidurePin.size
+   FroidurePin.sorted_at
+   FroidurePin.sorted_elements
+   FroidurePin.sorted_position
+   FroidurePin.suffix
+   FroidurePin.to_sorted_position
+
+Full API
+--------
+
+.. autoclass:: FroidurePin
+   :members:
+   :class-doc-from: init
+   :special-members: __init__
```

Comment:
```suggestion
```
Since  we have `:class-doc-from: init`, we don't need `:special-members: __init__`.